pkg/controllers/serving_controller.go

```diff
@@ -136,28 +136,35 @@ func (r *ServingReconciler) createOrUpdateServing(s *openfunction.Serving) (ctrl
 	ksvc.Name = fmt.Sprintf("%s-%s", s.Name, "ksvc")
 	ksvc.Namespace = s.Namespace
 
-	if err := r.Delete(r.ctx, &ksvc); util.IgnoreNotFound(client.IgnoreNotFound(err)) != nil {
+	status := openfunction.ServingStatus{Phase: openfunction.ServingPhase, State: openfunction.Launching}
```

Comment:
It's safer to modify the begining of createOrUpdateServing ?

```
if s.Status.Phase == openfunction.ServingPhase && ( s.Status.State == openfunction.Launched || s.Status.State == openfunction.Launching )
{ return ctrl.Result{}, nil	}
```

Reply:
There is a bug in deleting the serving before the status is updated. 

When the serving is created successfully, it will update the serving state to Launched. It takes a certain amount of time for the informer to synchronize data. The next time the serving Reconcile is triggered, the serving state gets from the informer may still be Launching. 

If the deletion is performed first, the ksvc will be deleted, and then update the serving status to Launching. At this time, the synchronization of the informer will cause the update status to fail, and Reconcile will retry. At this time, the serving status gets from the cache is Launched, and the ksvc will not be created.

If the update is performed first, firstly update the state to Launching, if the update is successful, then the ksvc will be recreated, else if the update is failed, the ksvc will not be deleted, and Reconcile will retry. At this time, the serving status gets from the cache is Launched, nothing to do.